docs/performance.md

```diff
@@ -3,4 +3,136 @@ uri: "/docs/performance/"
+
+## Pagination
+
+WPGraphQL implements Relay-style cursor-based pagination, which is more efficient than offset-based pagination for large datasets.
+
+Best practices for pagination:
+- Use reasonable `first`/`last` limits (10-100 items per request)
+- Implement infinite scroll (or load more buttons) rather than loading all items at once
+- Utilize cursor-based navigation with `first & after`/`last & before` parameters
+
+```graphql
+query GetPosts{
+  posts(first: 10, after: "cursor") {
+    pageInfo {
+      hasNextPage
+      endCursor
+      hasPreviousPage
+      startCursor
+    }
+    nodes {
+      id
+      title
+    }
+  }
+}
+```
+
+## Caching
+
+### WPGraphQL Smart Cache
+
+WPGraphQL provides an extension plugin, WPGraphQL Smart Cache, which is designed to work with network/edge caching (i.e. Varnish, Cloudflare, Fastly). 
+
+WPGraphQL returns an X-GraphQL-Keys header that contains tags relevant to the executed query. 
+
+Hosts that support it, will cache the GraphQL response and tag the cached document with the tags returned in the X-GraphQL-Keys header.
+
+WPGraphQL Smart Cache will then listen for events such as posts (of any post type, and terms, users, comments, etc) being published, updated or deleted, and will emit an event indicating which "tags" have been impacted, allowing cached documents with the corresponding tag to be purged from the cache.
+
+Learn more about [WPGraphQL Smart Cache](https://github.com/wp-graphql/wp-graphql-smart-cache). 
+
+### GET vs POST Requests
+
+While GraphQL typically uses POST requests, WPGraphQL supports GET requests which can be beneficial for:
+- CDN caching
+- Browser caching
+- Proxy caching
+- WPGraphQL Smart Cache
+
+To use GET requests:
+1. Make sure your client sends the request as GET instead of POST
+2. Ensure your queries are idempotent (read-only) - mutations will not work over GET.
+3. Be mindful of URL length limitations
+  - WPGraphQL Smart Cache has a "Persisted Queries" feature that can help overcome this limitation.
+
+## Request Strategies
+
+### Multiple Small Requests vs. Single Large Request
+
+Consider these approaches:
+
+**Multiple Small Requests:**
+- ✅ Better cache utilization
+- ✅ Parallel loading possible
+- ✅ Smaller payload per request
+- ❌ Multiple network roundtrips
+- ❌ More HTTP overhead
+
+**Single Large Request:**
+- ✅ Single network roundtrip
+- ✅ Less HTTP overhead
+- ❌ All-or-nothing cache invalidation
+- ❌ Larger initial payload
+- ❌ Potentially slower time-to-first-byte
+
+Choose based on your specific use case:
+- Use multiple requests for independent components that can load in parallel
+- Use single requests for tightly coupled data needed all at once
+
+## Performance Monitoring
+
+Monitor your GraphQL performance using:
+- Browser Developer Tools Network tab
+- WPGraphQL Query Logging and Trace data (enabled under WPGraphQL > Settings)
+- Server-side logging
+- Application Performance Monitoring (APM) tools
+
+## Additional Tips
```

Comment:
_Maybe_ also suggest adding a global `id` fields to your queries/fragments, since most query libraries use it for query deduplication/caching.